src/tribler-core/tribler_core/modules/libtorrent/download.py

```diff
@@ -75,6 +76,7 @@ def __init__(self, session, tdef):
 
         for alert_type, alert_handler in alert_handlers.items():
             self.register_alert_handler(alert_type, alert_handler)
+        self.stream = Stream(self)
```

Comment:
Maybe set the `stream` attribute only when it's needed? This would prevent stuff like `__prepare_coro` in the `Stream` class.

Reply:
i think if I add 2 methods to the download module it is possible.
1) a getter method to instantiate stream class to the download's self when needed.
2) a utility method to check if download is in streaming mode

I also need to make sure download.stream is only requested only for streaming purposes because the getter will instantiate it lazily when actually needed.

I think this will utilize ram much more effieciently get rid of stream instance overhead.

Reply:
I first thought maybe the `Download` class called `Stream.enable()`, but considering that it doesn't, you could also just keep it as is. The more we can keep the streaming logic out of `Download`, the better.